test/e2e/metrics_test.go

```diff
@@ -1,77 +1,120 @@
+// Package e2e contains end-to-end tests to verify that the metrics endpoints
+// for both components. Metrics are exported and accessible by authorized users through
+// RBAC and ServiceAccount tokens.
+//
+// These tests perform the following steps:
+// 1. Create a ClusterRoleBinding to grant necessary permissions for accessing metrics.
+// 2. Generate a ServiceAccount token for authentication.
+// 3. Deploy a curl pod to interact with the metrics endpoint.
+// 4. Wait for the curl pod to become ready.
+// 5. Execute a curl command from the pod to validate the metrics endpoint.
+// 6. Clean up all resources created during the test, such as the ClusterRoleBinding and curl pod.
+//
+//nolint:gosec
 package e2e
 
 import (
 	"bytes"
 	"io"
 	"os/exec"
+	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/require"
+
+	"github.com/operator-framework/operator-controller/test/utils"
 )
 
-// nolint:gosec
 // TestOperatorControllerMetricsExportedEndpoint verifies that the metrics endpoint for the operator controller
-// is exported correctly and accessible by authorized users through RBAC and a ServiceAccount token.
-// The test performs the following steps:
-// 1. Creates a ClusterRoleBinding to grant necessary permissions for accessing metrics.
-// 2. Generates a ServiceAccount token for authentication.
-// 3. Deploys a curl pod to interact with the metrics endpoint.
-// 4. Waits for the curl pod to become ready.
-// 5. Executes a curl command from the pod to validate the metrics endpoint.
-// 6. Cleans up all resources created during the test, such as the ClusterRoleBinding and curl pod.
 func TestOperatorControllerMetricsExportedEndpoint(t *testing.T) {
-	var (
-		token   string
-		curlPod = "curl-metrics"
-		client  = ""
-		clients = []string{"kubectl", "oc"}
+	client := utils.FindK8sClient(t)
+	config := NewMetricsTestConfig(
+		t, client,
+		"control-plane=operator-controller-controller-manager",
+		"operator-controller-metrics-reader",
+		"operator-controller-metrics-binding",
+		"operator-controller-controller-manager",
+		"oper-curl-metrics",
+		"https://operator-controller-service.NAMESPACE.svc.cluster.local:8443/metrics",
 	)
 
-	t.Log("Looking for k8s client")
-	for _, c := range clients {
-		// Would prefer to use `command -v`, but even that may not be installed!
-		err := exec.Command(c, "version", "--client").Run()
-		if err == nil {
-			client = c
-			break
-		}
-	}
-	if client == "" {
-		t.Fatal("k8s client not found")
+	config.createMetricsClusterRoleBinding()
+	token := config.getServiceAccountToken()
+	config.createCurlMetricsPod()
+	config.validate(token)
+	defer config.cleanup()
```

Comment:
Maybe add a `config.run()` that encapsulates these calls since they look the same in both tests?

Reply:
Done !!!
Good call.